backend/app/services/external/aws/s3.py

```
@@ -0,0 +1,50 @@
+        except ClientError as e:
+            logger.error(f"Error generating pre-signed upload URL: {e}")
+            raise BadRequestException(message=f"Error generating pre-signed upload URL: {e}")
+
+    def generate_download_url(self, file_key: str, expiration: int = 3600):
+        #"""Generate a pre-signed URL for downloading a file from S3."""
+        try:
+            presigned_url = self.s3_client.generate_presigned_url('get_object',
+                                                                    Params={'Bucket': self.bucket_name,
+                                                                            'Key': file_key},
+                                                                    ExpiresIn=expiration)
+            return presigned_url
```

Ruff (0.14.10)

Consider moving this statement to an `else` block

(TRY300)